runtime/lua/vim/fs.lua

```diff
@@ -275,6 +280,24 @@ function M.find(names, opts)
   local stop = opts.stop
   local limit = opts.limit or 1
 
+  if opts.dfs then
+    local matches = {}
+    if type(names) == 'function' then
+      error("Can't use dfs with functions")
+    end
+    for _, name in ipairs(names) do
+      opts.dfs = false
+      local match = vim.fs.find(name, opts)
```

Comment:
I'm not sure implementing this in terms of recursively calling `vim.fs.find` is the way to go. I think we should implement the DFS behavior directly. Will probably need to refactor the separate DFS and BFS behavior into separate functions and then have `vim.fs.find` call the appropriate one to avoid this function getting too complicated.

Reply:
Couldn't get it to work. Unsure if it'll be any easier than this, at least with this approach the two methods are completely separated which helps make it easier.

Reply:
I will make you regret this. You will get what you wish for. Prepare